test/foundry/OffchainAssetReceiptVaultFactory.t.sol

```diff
@@ -0,0 +1,57 @@
+pragma solidity 0.8.17;
+
+import "../../contracts/vault/receipt/ReceiptVault.sol";
+import "../../contracts/vault/receipt/ReceiptFactory.sol";
+
+import "forge-std/Test.sol";
+import "forge-std/console.sol";
+
+import "../../contracts/vault/offchainAsset/OffchainAssetReceiptVaultFactory.sol";
+import "../../contracts/vault/offchainAsset/OffchainAssetReceiptVault.sol";
+
+contract OffChainAssetReceiptVaultFactoryTest is Test {
+    OffchainAssetReceiptVaultFactory factory;
+    OffchainAssetVaultConfig offchainAssetVaultConfig;
+    VaultConfig vaultConfig;
+    OffchainAssetReceiptVault implementation;
+    ReceiptFactory receiptFactory;
+    ReceiptVaultFactoryConfig factoryConfig;
+    OffchainAssetReceiptVault vault;
+    address alice;
+
+    function setUp() public {
```

Comment:
this is going to be the same for a lot of tests, maybe we should make an abstract contract that they can all inherit